server/controllers/userController.js

```diff
@@ -211,6 +213,36 @@ class UserController {
     );
     return util.successStatus(res, 200, 'Password reset successfully');
   }
+  static async assignUserRole(req, res) {
+    const { id } = req.user ? req.user : req.query;
+    const { email, role } = req.body;
+    if( !id ) { return util.errorStatus(res, 401, 'Not Authorized'); }
+    try {
+        const superAdmin = await models.Users.findByPk(id);
+        if(!superAdmin || superAdmin.role !== 'super_admin') { return util.errorStatus(res, 401, 'Not Authorized'); }
```

Comment:
I think the Authorization should be done in a separate file, to avoid repetition.

Reply:
+1